lib/sport_ngin_aws_auditor/instance_helper.rb

```diff
@@ -129,7 +163,8 @@ def add_additional_data(ris_region, instances_with_tag, ignored_instances, diffe
     def compare(instances, ignore_instances_regexes, client, klass)
       ignored_instances, instances_with_tag, instance_hash = sort_through_instances(instances, ignore_instances_regexes)
       ris_region, ris_hash = sort_through_RIs(client)
-      differences = measure_differences(instance_hash, ris_hash)
+      # ris_region is empty for rds and cache, and ris_hash is empty for ec2 (by default)
+      differences = measure_differences(instance_hash, ris_hash, ris_region, klass)
```

Comment:
Do these changes take into account when the RIs may not be region-based, but rather zone-based? We should make sure we test with that.

Reply:
Yes. Zone-based RIs can only be used for specific zones but region-based RIs can be used for any zones. So this change will
1. Match all zone-based RIs with the corresponding AZ specific instances. 
2. Then, use region-based RIs to cover any left over instances that don't have any/enough corresponding zone specific RIs.

Reply:
Ah beautiful!